### PR TITLE
Adding uniqueness constraint on queue name

### PR DESCRIPTION
**Pull request by thehamzarocks**

### CODE REVIEW

File: complete/src/main/java/com/example/restservice/dao/Queue.java [hunk 1]
```diff
@@ -1,18 +1,15 @@
 package com.example.restservice.dao;
 
 import java.util.List;
-import javax.persistence.CascadeType;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
-import javax.persistence.OneToMany;
-import javax.persistence.Table;
+import javax.persistence.*;
```
Comment (thehamzarocks) [inline complete/src/main/java/com/example/restservice/dao/Queue.java:4]: Looks like IntelliJ automatically changed this.
Do we need project-wide editor settings?

File: complete/src/main/java/com/example/restservice/service/QueueService.java [hunk 2]
```diff
@@ -24,8 +25,14 @@
   @Autowired private UserService userService; // TODO remove
 
   public CreateQueueResponse createQueue(CreateQueueRequest createQueueRequest) {
-    var queue = queueRepository.save(new Queue(createQueueRequest.getQueueName()));
-    return new CreateQueueResponse(queue.getQueueName(), queue.getQueueId());
+    try {
```
Comment (thehamzarocks) [inline complete/src/main/java/com/example/restservice/service/QueueService.java:28]: I'm not familiar with the latest java features. Any cleaner way of doing this?

Comment (daltonfury42) [reply complete/src/main/java/com/example/restservice/service/QueueService.java:28]: This looks good to me, just that we don't need the second catch-all. We can let that exception flow down, and then spring would catch it an respond with an internal server error.

File: complete/src/main/java/com/example/restservice/dao/Queue.java (hunk 1 above)
Comment (daltonfury42) [reply complete/src/main/java/com/example/restservice/dao/Queue.java:4]: yeah, we will need to change that.

**Pull request merged**